addons/account/models/chart_template.py

```diff
@@ -48,6 +48,15 @@ def _remove_xml_id(xml_id):
             xml_id = old_tax.get_xml_id().get(old_tax.id)
             if xml_id:
                 _remove_xml_id(xml_id)
+
+        # the tax should be instanciated with the tag's country if there is one
+        template_rep_line_ids = template.invoice_repartition_line_ids + template.refund_repartition_line_ids
+        tag_countries = [line._get_tags_to_add().country_id.id for line in template_rep_line_ids if line._get_tags_to_add().country_id]
+        if tag_countries:
+            if tag_countries.count(tag_countries[0]) != len(tag_countries):
+                raise ValidationError(_("Couldn't create tax with template repartition lines having tags with different countries."))
+            template_vals["country_id"] = tag_countries[0]
```

Comment:
Code can be a little bit more elegant and optimized using set comprehension
```suggestion
        tag_countries = {line._get_tags_to_add().country_id.id for line in template_rep_line_ids}
        if len(tag_countries) != 1:
            raise ValidationError(_("Couldn't create tax with template repartition lines having tags with different countries."))
        template_vals["country_id"] = tag_countries.pop()
```